### PR TITLE
cancel network deletion if pods exist

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2993.

### CODE REVIEW

**Review by tuommaki — APPROVED** [review]
LGTM :hearts:

File: service/controller/v3/resource/legacy/backoff.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+const (
+	LongMaxWait  = 30 * time.Minute
+	ShortMaxWait = 2 * time.Minute
+)
+
+const (
+	LongMaxInterval  = 60 * time.Second
+	ShortMaxInterval = 5 * time.Second
```
Comment (kopiczko) [inline service/controller/v3/resource/legacy/backoff.go]: Are Long versions used?

File: service/controller/v3/key/key.go [hunk 2]
```diff
@@ -53,6 +53,10 @@ func HostPrivateNetwork(customObject v1alpha1.FlannelConfig) string {
 	return customObject.Spec.Bridge.Spec.PrivateNetwork
 }
 
+func IsInDeletionState(customObject v1alpha1.FlannelConfig) bool {
+	return customObject.GetDeletionTimestamp() != nil
+}
```
Comment (kopiczko) [inline service/controller/v3/key/key.go]: I would call it `IsDeleted` to shorten the name, and that's what actually happen.

**Review by kopiczko — APPROVED** [review]

File: service/controller/v3/resource/legacy/backoff.go (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/v3/resource/legacy/backoff.go]: Here not right now. This is actually copy/paste from e2e-harness where we use both. I do not mind though. At some point we may want to put this into a common place. Uncertain.

File: service/controller/v3/key/key.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/v3/key/key.go]: Done.

**Pull request merged**